tests/Test/Xrefcheck/AnchorsSpec.hs

```diff
@@ -5,18 +5,37 @@
 
 module Test.Xrefcheck.AnchorsSpec where
 
-import Test.Hspec (Spec, describe, it)
+import qualified Data.ByteString.Lazy as BSL 
+import Test.Hspec (Spec, describe, it, shouldBe)
 import Test.QuickCheck ((===))
 
-import Xrefcheck.Core (Flavor (..), headerToAnchor)
+import Xrefcheck.Core
+import Xrefcheck.Scanners.Markdown
 
 checkHeaderConversions
   :: HasCallStack
   => Flavor -> [(Text, Text)] -> Spec
 checkHeaderConversions fl suites =
-  describe (show fl) $
+  describe (show fl) $ do
     forM_ suites $ \(a, b) ->
       it (show a <> " == " <> show b) $ headerToAnchor fl a === b
+    it "Non-stripped header name should be stripped" $ do
+      fi <- getFI "tests/markdowns/without-annotations/non_stripped_spaces.md"
+      getAnchors fi `shouldBe` [ case fl of GitHub -> "header--with-leading-spaces"
+                                            GitLab -> "header-with-leading-spaces"
+                               ]
+  where
+    parse :: FilePath -> IO (Either Text FileInfo)
+    parse path =
+      parseFileInfo MarkdownConfig{ mcFlavor = fl } . decodeUtf8 <$> BSL.readFile path
+
+    getFI :: FilePath -> IO FileInfo
+    getFI path =
+      let errOrFI = parse path
+      in either error id <$> errOrFI
```

Comment:
Oh, I suspect that this is copied from another place.

Could you please create an `Util` module in tests and place this reading function there?

Reply:
Absolutely.

Reply:
Hmm, on another note, the definitions of `parse` slightly differ in the respective test modules: in `AnchorSpec` `parse` depends on the `Flavor` in the local scope, but in `IgnoreAnnotationSpec` it doesn't and uses the default GitHub configuration. May I use the default config in `AnchorSpec` as well (for the test's primary purpose is to ensure that there are no leading hyphens, which works for both `Flavor`s) to then put `parse`, `getAnchors` and `getFI` in the `Util` module?

Reply:
I personally would make the function accept `Flavor` explicitly because generally there is no sense of "default" flavor. 

The one who writes his own test should decide, whether can he test that on just one arbitrary flavor or try every possible flavor, and it's good if we force him to take this decision.